### DISCUSSION TITLE
In AOT module-->Exception: native stack overflow

### DISCUSSION THREAD

**Discussion by subhakr** [bytecodealliance/wasm-micro-runtime#3772] [Q&A]
### Subject of the issue
I have compiled Coremark-pro to wasm using wasi-sdk and running the .wasm file using wasm-micro-runtime(WAMR)
there are 9 workloads in coremark-pro in that parser-125k workload compiled to AOT, it is not running.
I am using the iwasm binary from wamr 
I have compiled Coremark-pro to wasm using wasi-sdk and running the .wasm
### Your environment
* Host OS : Linux 22.04
* WAMR version: 2.1.1
* Wasi-sdk compiler: clang-18
*  cpu architecture: x86 
*  running mode: AOT 

### Steps to reproduce
```
$ /coremark-pro_t1/builds/linux64/gcc64/bin$ ../../../../iwasm  --heap-size=3500000  parser-125k.wasm 
-  Info: Starting Run...
-- Workload:parser-125k=780641437
-- parser-125k:time(ns)=83
-- parser-125k:contexts=1
-- parser-125k:iterations=1
-- parser-125k:time(secs)=   0.083
-- parser-125k:secs/workload=   0.083
-- parser-125k:workloads/sec= 12.0482
```

when running the same for AOT 
`$ /coremark-pro_t1/builds/linux64/gcc64/bin$ ../../../../iwasm -v=5 - parser-125k.aot
`

###Tell us how to reproduce this error
Run the attached bin files one is .wasm and other is .aot with the  iwasm
### Expected behavior

for both wasm and AOT i shoud get the result or score as below
```
-  Info: Starting Run...
-- Workload:parser-125k=780641437
-- parser-125k:time(ns)=83
-- parser-125k:contexts=1
-- parser-125k:iterations=1
-- parser-125k:time(secs)=   0.083
-- parser-125k:secs/workload=   0.083
-- parser-125k:workloads/sec= 12.0482
```

### Actual behavior

In linker flags like below  i am using values.
```
LINKER_FLAGS += $(EM_FLAGS) -pthread -O3 \
    -Wl,--shared-memory \
    -Wl,--allow-undefined,--no-check-features \
    -Wl,--export=__heap_base,--export=__data_end \
    -Wl,--export=malloc -Wl,--export=free \
    -Wl,--import-memory -Wl,--export-memory \
    -Wl,--max-memory=2147483648 -z stack-size=4500000

```
**method1:-**
If stack size is lesser than heap wasm module is working and generating score.
and with same values i ahve tried to run AOT module which is converted from wasm module.
with AOT module i am facing issue like **Exception: native stack overflow**
**method2:-**
Here i have increased stack size more than heap then wasm module itself is not generating score.
its giving error like 
```
[00:02:15:240 - 767A12A77740]: warning: allocate 3900000 bytes memory failed
[00:02:15:240 - 767A12A77740]: thread manager error: failed to allocate aux stack space for new thread

```

* This is the behaviour of less stack size than heap with AOT module

```
[05:11:14:256 - 7F04A6AB2740]: Load module success.
 
[05:11:14:256 - 7F04A6AB2740]: Memory instantiate:
[05:11:14:256 - 7F04A6AB2740]:   page bytes: 4558032, init pages: 1, max pages: 1
[05:11:14:256 - 7F04A6AB2740]:   data offset: 9456, stack size: 1048576
[05:11:14:256 - 7F04A6AB2740]:   heap offset: 1058032, heap size: 3500000
 
-  Info: Starting Run...
Exception: native stack overflow
```

* This is the behaviour of wasm module with more stack size than heap
```
-  Info: Starting Run...
[00:02:15:240 - 767A12A77740]: warning: allocate 3900000 bytes memory failed
[00:02:15:240 - 767A12A77740]: thread manager error: failed to allocate aux stack space for new thread
-- Workload:parser-125k=780641437
-- parser-125k:time(ns)=0
-- parser-125k:contexts=1
-- parser-125k:iterations=1
-- parser-125k:time(secs)=       0
-- parser-125k:secs/workload=       0
-- parser-125k:workloads/sec=     inf
```

**Comment by wenyongh** [comment]
Hi, did you build the iwasm from scratch, or just use the release iwasm binary? And the coremark wasm file isn't found, could you upload it?

**Reply by subhakr** [reply]
[parser-125k.zip](https://github.com/user-attachments/files/17430054/parser-125k.zip)
Hi you can find the attached file for parser workload.

and i have built iwasm from sratch and enabled required flags in CmakeList.txt.

Thank you in advance

**Comment by wenyongh** [comment]
Hi, the default native stack size for child thread is a little small in the AOT mode, please enlarge it by manually changing the below line (e.g. change to `1280 * 1024`):
https://github.com/bytecodealliance/wasm-micro-runtime/blob/217ba3b10c41036f0e9727aa93c2a164dba35e5c/core/config.h#L434-L435
and build iwasm with lib-pthread support again:
```C
cd product-mini/platforms/linux
mkdir build && cd build
cmake .. -DWAMR_BUILD_LIB_PTHREAD=1
make -j
```
Or define the macro in your CMakeLists.txt, e.g. add `add_definitions(-DAPP_THREAD_STACK_SIZE_DEFAULT=1024000)`.

**Reply by subhakr** [reply]
It worked.
Thank you so much for your reply
